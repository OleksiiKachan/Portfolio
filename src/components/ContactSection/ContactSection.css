.contactSection {
    display: flex;
    flex-direction: column;
	align-items: center;
}

.contactSection__title {
    margin-bottom: 15px;
}

.contactSection__content {
    display: flex;
	justify-content: space-around;
	align-items: stretch;
	width: 80%;
}

.contactSection__contactColumn {
    width: 35%;
}