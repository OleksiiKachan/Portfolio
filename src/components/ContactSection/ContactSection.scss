@import '../../assets/style/variables';

.contactSection {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 5rem;
  background: url('https://res.cloudinary.com/oleksiikachan/image/upload/v1558643461/portfolio/images/contact_background.png');
  background-size: cover;
  background-position: center;
  background-attachment: fixed;
  box-shadow: inset 2px 4px 4px rgba(0, 0, 0, 0.25);
}

.contactSection__title {
  margin-bottom: 2rem;
  font-size: 1.5rem;
}

.contactSection__content {
  display: flex;
  justify-content: space-around;
  align-items: stretch;
  width: 80%;
}

.contactSection__contactColumn {
  width: 40%;
}

@media screen and (max-width: $breakpoint-tablet) {
  .contactSection {
    padding: 1rem 0;
    box-shadow: inset 2px 4px 4px rgba(0, 0, 0, 0.25);
  }

  .contactSection__title {
    font-size: 1.2rem;
  }

  .contactSection__content {
    flex-direction: column;
    align-items: center;
    width: 90%;
  }

  .contactSection__contactColumn {
    width: 100%;
    font-size: 0.8rem;
  }
}
