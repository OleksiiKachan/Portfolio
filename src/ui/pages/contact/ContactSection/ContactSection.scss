@import '../../../assets/style/variables';
@import '../../../assets/style/mixins';

.contactSection {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 5rem;
  background: url('https://res.cloudinary.com/oleksiikachan/image/upload/v1558643461/portfolio/images/contact_background.png');
  background-size: cover;
  background-position: center;
  background-attachment: fixed;
  box-shadow: inset 2px 4px 4px rgba(0, 0, 0, 0.25);

  @include respond(tablet) {
    padding: 1rem 0;
    box-shadow: inset 2px 4px 4px rgba(0, 0, 0, 0.1);
  }

  &__title {
    margin-bottom: 2rem;
  }

  &__content {
    display: flex;
    justify-content: space-around;
    align-items: stretch;
    width: 80%;

    @include respond(tablet) {
      flex-direction: column;
      align-items: center;
      width: 90%;
    }
  }

  &__contactColumn {
    width: 40%;

    @include respond(tablet) {
      width: 100%;
      font-size: 0.8rem;
    }
  }
}
