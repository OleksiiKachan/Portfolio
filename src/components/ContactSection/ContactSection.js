import React, { Component } from 'react';
import classNames from 'classnames';
import './ContactSection.scss';
import ContactForm from '../ContactForm';
import ContactInfoPanel from '../ContactInfoPanel';

/*
    <ContactSection contacts={} className='' />
*/
class ContactSection extends Component {
  render() {
    return (
      <div
        id="contact"
        className={classNames('contactSection', this.props.className)}
      >
        <h2 className={classNames('contactSection__title')}>Get In Touch</h2>
        <div className={classNames('contactSection__content')}>
          <div
            className={classNames(
              'contactSection__contactColumn',
              'contactSection__contactForm'
            )}
          >
            <ContactForm />
          </div>
          <div
            className={classNames(
              'contactSection__contactColumn',
              'contactSection__contactInfo'
            )}
          >
            <ContactInfoPanel contacts={this.props.contacts} />
          </div>
        </div>
      </div>
    );
  }
}

export default ContactSection;
