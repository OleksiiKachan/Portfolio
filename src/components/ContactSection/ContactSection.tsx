import React, { Component } from 'react';
import classNames from 'classnames';
import './ContactSection.scss';
import ContactForm from './ContactForm';
import ContactInfoPanel from './ContactInfoPanel';

/*
    <ContactSection contacts={} className='' />
*/
class ContactSection extends Component<any, any> {
  render() {
    return (
      <div
        id="contact"
        className={classNames('contactSection', this.props.className)}
      >
        <h2 className={classNames('contactSection__title')}>Get In Touch</h2>
        <div className={classNames('contactSection__content')}>
          <div className={classNames('contactSection__contactColumn')}>
            <ContactForm />
          </div>
          <div className={classNames('contactSection__contactColumn')}>
            <ContactInfoPanel contacts={this.props.contacts} />
          </div>
        </div>
      </div>
    );
  }
}

export default ContactSection;
