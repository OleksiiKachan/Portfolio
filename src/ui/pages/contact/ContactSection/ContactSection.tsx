import React from 'react';
import classNames from 'classnames';
import './ContactSection.scss';
import ContactForm from './ContactForm/ContactForm';
import ContactInfoPanel from './ContactInfoPanel/ContactInfoPanel';

/*
    <ContactSection
      contacts={}
      className=''
    />
*/

type PropsType = {
  contacts: {
    phone: { caption: string; url: string };
    email: { caption: string; url: string };
    location: { caption: string; url: string };
  };
  className?: string;
};

export default ({ contacts, className }: PropsType) => {
  return (
    <div id="contact" className={classNames('contactSection', className)}>
      <h2 className={classNames('contactSection__title')}>Get In Touch</h2>
      <div className={classNames('contactSection__content')}>
        <div className={classNames('contactSection__contactColumn')}>
          <ContactForm />
        </div>
        <div className={classNames('contactSection__contactColumn')}>
          <ContactInfoPanel contacts={contacts} />
        </div>
      </div>
    </div>
  );
};
