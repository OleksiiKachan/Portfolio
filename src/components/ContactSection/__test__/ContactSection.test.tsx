import React from 'react';
import { shallow } from 'enzyme';
import ContactSection from '..';

describe('ContactSection', () => {
  it('Snapshot', () => {
    const component = shallow(
      <ContactSection
        contacts={{
          phone: 'phone number',
          email: 'email',
          location: 'location',
        }}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
