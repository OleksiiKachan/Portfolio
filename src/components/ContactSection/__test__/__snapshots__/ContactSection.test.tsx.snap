// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactSection Snapshot 1`] = `
<div
  className="contactSection"
  id="contact"
>
  <h2
    className="contactSection__title"
  >
    Get In Touch
  </h2>
  <div
    className="contactSection__content"
  >
    <div
      className="contactSection__contactColumn"
    >
      <_default />
    </div>
    <div
      className="contactSection__contactColumn"
    >
      <_default
        contacts={
          Object {
            "email": "email",
            "location": "location",
            "phone": "phone number",
          }
        }
      />
    </div>
  </div>
</div>
`;
